internal/service/datazone/domain.go

```diff
@@ -165,29 +170,37 @@ func (r *domainResource) Create(ctx context.Context, request resource.CreateRequ
 	const (
 		timeout = 30 * time.Second
 	)
-	output, err := tfresource.RetryWhenAWSErrCodeContains(ctx, timeout, func(ctx context.Context) (*datazone.CreateDomainOutput, error) {
+	out, err := tfresource.RetryWhenAWSErrCodeContains(ctx, timeout, func(ctx context.Context) (*datazone.CreateDomainOutput, error) {
 		return conn.CreateDomain(ctx, &input)
 	}, errCodeAccessDenied)
 
 	if err != nil {
 		response.Diagnostics.AddError(fmt.Sprintf("creating DataZone Domain (%s)", name), err.Error())
-
 		return
 	}
 
-	// Set values for unknowns.
-	data.ARN = fwflex.StringToFramework(ctx, output.Arn)
-	data.ID = fwflex.StringToFramework(ctx, output.Id)
-	data.PortalURL = fwflex.StringToFramework(ctx, output.PortalUrl)
-	data.DomainVersion = fwtypes.StringEnumValue[awstypes.DomainVersion](output.DomainVersion)
+	data.ID = fwflex.StringToFramework(ctx, out.Id)
 
-	if _, err := waitDomainCreated(ctx, conn, data.ID.ValueString(), r.CreateTimeout(ctx, data.Timeouts)); err != nil {
+	output, err := waitDomainCreated(ctx, conn, data.ID.ValueString(), r.CreateTimeout(ctx, data.Timeouts))
+	if err != nil {
 		response.State.SetAttribute(ctx, path.Root(names.AttrID), data.ID) // Set 'id' so as to taint the resource.
 		response.Diagnostics.AddError(fmt.Sprintf("waiting for DataZone Domain (%s) create", data.ID.ValueString()), err.Error())
 
 		return
 	}
 
+	// Do not set single sign on in state if it was null and response is DISABLED as this is equivalent.
+	if output != nil {
+		if output.SingleSignOn.Type == awstypes.AuthTypeDisabled && data.SingleSignOn.IsNull() {
```

Comment:
To avoid a potential nil pointer dereference.

```suggestion
		if output.SingleSignOn != nil && output.SingleSignOn.Type == awstypes.AuthTypeDisabled && data.SingleSignOn.IsNull() {
```